borg/helpers.py

```diff
@@ -1107,9 +1114,17 @@ class FakeArchive:
             b'mode': 0, b'path': '', b'user': '', b'group': '', b'mtime': 0,
             b'uid': 0, b'gid': 0,
         }).keys())
-        keys.sort()
-        keys.sort(key=str.isupper)
-        return keys
+        return sorted(sorted(keys), key=str.isupper)
```

Comment:
sort is stable, but I just remembered that tuples also work for keys, so that's simpler. Will include that in the next patch rev.